dapps/assets/src/components/digitaltwins/AddDigitalTwin.ts

```diff
@@ -116,24 +75,30 @@ class AddDigitalTwinComponent extends mixins(EvanComponent) {
    * @param files
    */
   async handleFileUpload (files: UIContainerFile[]) {
-    // reset when file was deleted
+    // skip when file was deleted
     if (!files[0]) {
-      this.template = null;
-
       return;
     }
 
+    const rollBackTemplate = JSON.stringify(this.template);
+
     this.template = await this._blobToObj(files[0].blob) as DigitalTwinTemplate;
 
-    if (!this.name) {
-      this.name = this.getLocalizedTemplateEntry(this.template, 'name');
-    }
+    // @ts-ignore // TODO: update dts of validateDescription()
+    this.templateErrors = this.runtime.description.validateDescription(this.template);
 
-    if (!this.description) {
-      this.description = this.getLocalizedTemplateEntry(this.template, 'description');
+    if (this.templateErrors && this.templateErrors.length > 0) {
+      this.template = JSON.parse(rollBackTemplate);
+
+      return;
     }
 
-    this.selectedTemplate = null;
+    this._setDefaults();
+    this._addToPresetTemplates(this.template, files[0].name);
+    this.selectedTemplate = files[0].name;
+
+    // ugly hack to wait till underlying select component received new props
+    window.setTimeout(() => this.$forceUpdate(), 200);
```

Comment:
Ist this also working?

```suggestion
    this.$nextTick(() => this.$forceUpdate(), 200);
```

Reply:
nope, but resolved with way better nextTicket on select component, thanks for the hint! 🎉